--- a/subprojects/wlroots/include/wlr/types/wlr_switch.h
+++ b/subprojects/wlroots/include/wlr/types/wlr_switch.h
@@ -36,6 +36,7 @@
 enum wlr_switch_type {
 	WLR_SWITCH_TYPE_LID,
 	WLR_SWITCH_TYPE_TABLET_MODE,
+	WLR_SWITCH_TYPE_KEYPAD_SLIDE,
 };
 
 enum wlr_switch_state {
--- a/subprojects/wlroots/backend/libinput/switch.c
+++ b/subprojects/wlroots/backend/libinput/switch.c
@@ -34,6 +34,9 @@
 		wlr_event.switch_type = WLR_SWITCH_TYPE_LID;
 		break;
 	case LIBINPUT_SWITCH_TABLET_MODE:
+	case LIBINPUT_SWITCH_KEYPAD_SLIDE:
+		wlr_event.switch_type = WLR_SWITCH_TYPE_KEYPAD_SLIDE;
+		break;
 		wlr_event.switch_type = WLR_SWITCH_TYPE_TABLET_MODE;
 		break;
 	}
